systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -253,4 +258,182 @@ void testKafkaNodePoolFeatureGate(ExtensionContext extensionContext) {
+                .withNamespace(testStorage.getNamespaceName())
+            .endMetadata()
+            .editOrNewSpec()
+                .addToRoles(ProcessRoles.BROKER)
+                .withStorage(kafkaCr.getSpec().getKafka().getStorage())
+                .withJvmOptions(kafkaCr.getSpec().getKafka().getJvmOptions())
+                .withResources(kafkaCr.getSpec().getKafka().getResources())
+            .endSpec()
+            .build();
+
+        resourceManager.createResourceWithWait(extensionContext,
+            kafkaNodePoolCr,
+            kafkaCr);
+
+        LOGGER.info("Creating KafkaTopic: {}/{}", testStorage.getNamespaceName(), testStorage.getTopicName());
+        resourceManager.createResourceWithWait(extensionContext, KafkaTopicTemplates.topic(testStorage).build());
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResourceWithWait(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+        // increase number of kafka replicas in KafkaNodePool
+        LOGGER.info("Modifying KafkaNodePool: {}/{} by increasing number of Kafka replicas from '3' to '5'", testStorage.getNamespaceName(), kafkaNodePoolName);
+        KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(kafkaNodePoolName, kafkaNodePool -> {
+            kafkaNodePool.getSpec().setReplicas(nodePoolIncreasedKafkaReplicaCount);
+        }, testStorage.getNamespaceName());
+
+        StrimziPodSetUtils.waitForAllStrimziPodSetAndPodsReady(
+            testStorage.getNamespaceName(),
+            KafkaResource.getStrimziPodSetName(testStorage.getClusterName(), kafkaNodePoolName),
+            KafkaResources.kafkaStatefulSetName(testStorage.getClusterName()),
+            nodePoolIncreasedKafkaReplicaCount
+        );
+
+        LOGGER.info("Producing and Consuming messages with clients: {}, {} in Namespace {}", testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName());
+        resourceManager.createResourceWithWait(extensionContext,
+            clients.producerStrimzi(),
+            clients.consumerStrimzi()
+        );
+        ClientUtils.waitForClientsSuccess(testStorage);
+
+        LOGGER.info("Changing FG env variable to disable Kafka Node Pools");
+        coEnvVars = kubeClient().getDeployment(clusterOperator.getDeploymentNamespace(), Constants.STRIMZI_DEPLOYMENT_NAME).getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+        coEnvVars.stream().filter(env -> env.getName().equals(Environment.STRIMZI_FEATURE_GATES_ENV)).findFirst().get().setValue("-KafkaNodePools");
+
+        Deployment coDep = kubeClient().getDeployment(clusterOperator.getDeploymentNamespace(), Constants.STRIMZI_DEPLOYMENT_NAME);
```

Comment:
I think it might to worth considering some helper method, which doing such logic. Because AFAIK we have multiple test cases which tried the following :
1. Disable FG
2. Enable FG 
and this follows the waiting for the rolling update and many more operations which seems that we are repeating in each test case and can be generalized by some helper method I assume. WDYT?